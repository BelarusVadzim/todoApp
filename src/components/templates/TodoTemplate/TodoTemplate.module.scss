@import "~/src/variables.scss";

.background {
  position: fixed;
  height: 100vh;
  width: 100vw;
  z-index: -1;
}

.backgroudImage {
  position: fixed;
  top: 0;
  background-image: linear-gradient(to right, rgb(255 33 248 / 73%),  rgb(49 80 255 / 73%)),
    url("~/public/nature2.jpg");
  width: 100vw;
  height: 17rem;
}

.page {
  height: 100vh;
  width: 100vw;
  position: fixed;
}

.main {
  position: relative;
  margin-right: auto;
  margin-left: auto;
}

