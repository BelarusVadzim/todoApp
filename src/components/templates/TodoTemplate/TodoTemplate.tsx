import React from 'react';
import { TodoList } from 'components/organisms';
import style from './TodoTemplate.module.scss';

const TodoTemplate: React.FC = () => (
  <>
    <div className={style.background}></div>
    <div className={style.backgroudImage}></div>
    <div className={style.page}>
      <div className={style.main}>
        <TodoList />
      </div>
    </div>
  </>
);

export default TodoTemplate;
