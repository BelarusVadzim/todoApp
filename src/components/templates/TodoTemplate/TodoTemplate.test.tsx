import { render } from '@testing-library/react';
import TodoTemplate from './TodoTemplate';

jest.mock('./TodoTemplate.module.scss', () => ({
  background: 'backgroundStyle',
  backgroudImage: 'backgroudImageStyle',
  page: 'pageStyle',
  main: 'mainStyle',
}));

jest.mock('components/organisms', () => ({
  TodoList: jest.fn(() => (<div>TodoList</div>)),
}));

describe('<TodoTemplate />', () => {
  it('should render properly', () => {
    const component = render(<TodoTemplate />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
