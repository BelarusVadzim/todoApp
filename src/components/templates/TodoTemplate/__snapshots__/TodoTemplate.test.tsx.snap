// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTemplate /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="backgroundStyle"
  />
  <div
    class="backgroudImageStyle"
  />
  <div
    class="pageStyle"
  >
    <div
      class="mainStyle"
    >
      <div>
        TodoList
      </div>
    </div>
  </div>
</DocumentFragment>
`;
